src/main/java/com/strandls/cca/service/impl/CCADataServiceImpl.java

```diff
@@ -228,23 +233,38 @@ public AggregationResponse getCCADataList(HttpServletRequest request, UriInfo ur
 
 	@SuppressWarnings({ "unchecked", "rawtypes" })
 	@Override
-	public Map<String, Object> getCCADataAggregation(HttpServletRequest request, UriInfo uriInfo, boolean myListOnly)
+	public Map<String, Object> getCCADataAggregation(String query,HttpServletRequest request, UriInfo uriInfo, boolean myListOnly)
 			throws JsonProcessingException {
-		String userId = null;
+		
 		MultivaluedMap<String, String> queryParams = uriInfo.getQueryParameters();
+        
+        // Set the default value for the query parameter if not provided
+        if (query == null) {
+        	query = "";
+        }
+
+        List<String> fieldIds = ccaTemplateService.getFieldIds("", "");
+        List<String> valueFields=ccaTemplateService.getValueFields(fieldIds);
+        
+        // Add multiple fields to the search query
+        List<Bson> fieldQueries = new ArrayList<>();
+        for (String valueField : valueFields) {
+            Bson fieldQuery = Filters.regex(valueField, query, "i");
+            fieldQueries.add(fieldQuery);
+        }
+        
+        Bson searchQuery = Filters.or(fieldQueries);
+		
+		String userId = null;
 		if (myListOnly) {
 			CommonProfile profile = AuthUtil.getProfileFromRequest(request);
 			userId = profile.getId();
 		} else if (queryParams.containsKey(CCAConstants.USER_ID)) {
 			userId = queryParams.get(CCAConstants.USER_ID).get(0);
 		}
 
-		String language = queryParams.getFirst(CCAConstants.LANGUAGE);
-		if (language == null)
-			language = CCAConfig.getProperty(ApiConstants.DEFAULT_LANGUAGE);
-
-		AggregateIterable<Map> aggregation = ccaDataDao.getAggregation(uriInfo, userId);
```

Comment:
need to confirm this removal

Reply:
language variable is not used anywhere down the function, so removed it as it as causing a lint error